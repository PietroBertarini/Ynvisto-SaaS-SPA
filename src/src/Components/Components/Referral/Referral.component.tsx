import React from 'react';
import { Typography } from '@material-ui/core';
import YnvistoButton from '../YnvistoButton/YnvistoButton.component';
import { Main, useStyles } from './Referral.styles';

interface OwnProps {
  title: string;
  texts: string[];
  svg?: any;
  buttonTitle?: string;
  underButtonText?: string;
  onClick?: () => any;
}

const Referral: React.FC<OwnProps> = ({
  title, texts, svg, buttonTitle, underButtonText, onClick,
}: OwnProps) => {
  const classes = useStyles();

  return (
    <Main>
      <Typography className={classes.header}>{title}</Typography>
      <div className="svgArea">
        {svg}
      </div>
      {texts.map((text) => (
        <Typography className={classes.text} key={text}>{text}</Typography>
      ))}
      <div className="buttonArea">
        {buttonTitle && onClick && (
        <YnvistoButton
          title={buttonTitle}
          onClick={onClick}
          underButtonText={underButtonText}
        />
        )}
      </div>
    </Main>
  );
};

export default Referral;
