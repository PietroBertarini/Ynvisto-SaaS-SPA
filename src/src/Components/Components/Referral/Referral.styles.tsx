import styled from 'styled-components';
import { createStyles, makeStyles } from '@material-ui/core/styles';

export const Main = styled.div`
  width: 100%;
  margin-bottom: 10vh;
  display: flex; 
  flex-direction: column;

  .svgArea {
    text-align: center;
    height: 50vh;
  }

  .buttonArea {
    width: 80vw;
  }
`;

export const useStyles = makeStyles(() => createStyles({
  header: {
    color: '#FFFFFF',
    fontSize: '2.8vw',
    fontFamily: 'Zurich',
  },
  text: {
    color: '#fff',
    fontSize: '1.5vw',
    marginBottom: '20px',
  },
}));
