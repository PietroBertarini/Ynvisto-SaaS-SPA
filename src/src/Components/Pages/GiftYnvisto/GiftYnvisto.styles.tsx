import styled from 'styled-components';

export const Main = styled.div`
  width: 80vw;
  margin-left: 240px;
`;
