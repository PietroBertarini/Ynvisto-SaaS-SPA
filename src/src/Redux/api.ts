import axios, { AxiosResponse } from 'axios';

/**
 * Localhost YnvistoGateway api.
 */
const api = axios.create({
  baseURL: process.env.REACT_APP_API_ENDPOINT,
});

// /**
//  * Check if the one error have status 401.
//  * @param err: Caught error
//  */
// const isAuthError = (err: any): boolean => {
//   const axiosError = err as AxiosError;
//   // Whenever the erro is not from axios, we cannot try to repair it
//   if (!axiosError?.response) {
//     return false;
//   }
//
//   // Gets the error code from the axios response
//   const httpStatusCode = axiosError.response?.status;
//
//   // If the error code is Unauthorized, mark the error as Unauthorized error
//   return httpStatusCode === 401;
// };

/**
 * Always that callSafely catch error with status 401, will refresh
 * api token and call one more time.(Just 2 times), if this error occurs
 * again, than throw this error.
 * @param apiCall: Api with one method(get,post...)
 * @param authRetry: boolean to check if error 401 was caught one time.
 */
export function* callSafely<T>(apiCall: Promise<AxiosResponse<any>>, authRetry = false) {
  try {
    return yield apiCall;
  } catch (e) {
    if (!authRetry) {
      // TODO: Refresh token
      // yield callSafely(apiCall, true);
    }
    throw (e);
  }
}

export default api;
