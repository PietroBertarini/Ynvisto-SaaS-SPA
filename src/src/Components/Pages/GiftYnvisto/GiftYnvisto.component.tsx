import React, { useCallback, useEffect, useState } from 'react';
import Referral from '../../Components/Referral/Referral.component';
import { ReactComponent as Gift } from '../../../Assets/Images/gift.svg';
import { ReactComponent as Friendship } from '../../../Assets/Images/friendship.svg';
import { Main } from './GiftYnvisto.styles';
import { getLinkReferral, getInfoReferral, copyInvite } from './GiftYnvisto.utils';
import { IReferralResponse } from '../../../Redux/Services/Referral/Referral.api';
import Spinner from '../../Components/Spinner/spinner.component';

const GiftYnvisto: React.FC = () => {
  const [gettingInfo, setGettingInfo] = useState(true);
  const [redeemedInvites, setRedeemedInvites] = useState(0);
  const [remainingInvites, setRemainingInvites] = useState(0);

  useEffect(() => {
    async function fetchData() {
      const response = await getInfoReferral();
      setRedeemedInvites((response as IReferralResponse).redeemedInvites);
      setRemainingInvites((response as IReferralResponse).remainingInvites);
      setGettingInfo(false);
    }
    fetchData();
  }, [gettingInfo]);

  const title = 'Ajude outros investidores dando a Ynvisto de presente!';

  const youSavedText = (redeemedInvites > 0)
    ? `Você sabia que já salvou ${redeemedInvites} pessoas do inferno tributário brasileiro?` : '';
  const text = (remainingInvites > 0) ? [
    'Você conhece algum investidor que está precisando de uma ajuda com a tributação dos seus investimentos?',
    'Se sim, dê de presente 1 mês grátis de Ynvisto! Caso ele não tenha experimentado a plataforma ainda, '
  + 'o seu convite vale uma oferta especial de 1 mês grátis.',
    youSavedText,
  ] : [`Você já salvou ${redeemedInvites} pessoas do inferno tributário brasileiro!`,
    'Continue a ajudar mais investidores a simplificarem sua tributação na bolsa por meio do link abaixo.'];

  const buttonTitle = (remainingInvites > 0)
    ? 'Clique para copiar seu convite'
    : 'Clique para copiar sua recomendação';

  const underButtonText = (remainingInvites > 0)
    ? `Você tem direito à mais ${remainingInvites} convites com mês grátis`
    : '';

  const referralText = (remainingInvites > 0)
    ? 'Você recebeu um presente para conhecer a Ynvisto. Experimente gratuitamente por 30 dias '
  + 'uma vida na bolsa de valores sem se preocupar com os cálculos de sua tributação e aproveite de todas as normas '
  + 'para diminuir o valor de seu imposto.\n'
    : 'Você foi indicado para a Ynvisto. Isso significa que alguém quer te ajudar a lidar melhor com seu imposto na '
  + 'bolsa de valores, tendo menos trabalho e pagando menos. Experimente gratuitamente por 7 dias uma vida na '
  + 'bolsa de valores sem se preocupar com os cálculos de sua tributação e aproveite de todas as normas '
  + 'para diminuir o valor de seu imposto.\n https://www.ynvisto.com/';

  const getLink = useCallback(async () => {
    const response = await getLinkReferral();

    if (response !== '') {
      copyInvite(referralText + response);
    }
  }, [referralText]);

  if (gettingInfo) {
    return <Spinner />;
  }

  return (
    <Main>
      {!gettingInfo && (
      <Referral
        title={title}
        texts={text}
        svg={(remainingInvites > 0)
          ? (<Gift style={{ height: '50vh' }} />)
          : (<Friendship style={{ height: '50vh' }} />)}
        buttonTitle={buttonTitle}
        underButtonText={underButtonText}
        onClick={(remainingInvites > 0) ? getLink : () => copyInvite(referralText)}
      />
      )}
    </Main>
  );
};

export default GiftYnvisto;
