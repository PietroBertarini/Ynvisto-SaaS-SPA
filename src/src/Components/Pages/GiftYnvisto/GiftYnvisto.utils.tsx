import Swal from 'sweetalert2';
import { IApiError } from '../../../Redux/api.types';
import { multinesErrorToHtml } from '../../../Redux/Services/Api.utils';
import { getReferralLink, getReferrelInfo, IReferralResponse } from '../../../Redux/Services/Referral/Referral.api';

const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
const GET_LINK_DELAY = 600;

export const getLinkReferral = async () : Promise<string | IApiError> => {
  // Fires up a parallel delay that takes GET_LINK_DELAY milliseconds to complete
  const delayPromise = delay(GET_LINK_DELAY);
  const result = await getReferralLink();

  if ((result as IApiError).errorCode) {
    const linkError = result as IApiError;
    Swal.fire({
      icon: 'error',
      title: 'Um erro ocorreu ao tentar gerar o link de presente.',
      html: multinesErrorToHtml(linkError.errorMessage),
    });
  }

  // Ensures at least GET_LINK_DELAY milliseconds have passed. If API takes more than the minimum loading time, this
  // will be finished and resolve straight away. When API completes too fast, this ensures at least the configured time
  // will pass
  await delayPromise;

  return result;
};

export const getInfoReferral = async () : Promise<IReferralResponse | IApiError> => {
  const result = await getReferrelInfo();

  if ((result as IApiError).errorCode) {
    const linkError = result as IApiError;
    Swal.fire({
      icon: 'error',
      title: 'Um erro ocorreu ao tentar carregar esta pÃ¡gina.',
      html: multinesErrorToHtml(linkError.errorMessage),
    });
  }

  return result;
};

export const copyInvite = (textCopy: string) => {
  // Creates a dummy element, so we can copy it's value to clipboard
  const dummy = document.createElement('textarea');
  document.body.appendChild(dummy);
  dummy.value = textCopy;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);

  Swal.fire({
    icon: 'success',
    title: 'O convite foi copiado para seu clipboard.',
  });
};
