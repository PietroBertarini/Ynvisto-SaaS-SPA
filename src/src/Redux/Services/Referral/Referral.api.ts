import api from '../../api';
import { IApiError } from '../../api.types';
import { constructErrorFromApi } from '../Api.utils';

export interface IReferralResponse {
  redeemedInvites: number,
  remainingInvites: number,
}

export async function getReferralLink() : Promise<string | IApiError> {
  try {
    const result = await api.get('Referral/GetLink', {
      withCredentials: true,
    });

    return result.data;
  } catch (errorFromApi) {
    return constructErrorFromApi(errorFromApi);
  }
}

export async function getReferrelInfo() : Promise<IReferralResponse | IApiError> {
  try {
    const result = await api.get('Referral/GetReferralInfo', {
      withCredentials: true,
    });

    return result.data;
  } catch (errorFromApi) {
    return constructErrorFromApi(errorFromApi);
  }
}
