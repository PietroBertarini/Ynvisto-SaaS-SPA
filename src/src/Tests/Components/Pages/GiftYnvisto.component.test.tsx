import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import GiftYnvisto from '../../../Components/Pages/GiftYnvisto/GiftYnvisto.component';
import { IReferralResponse } from '../../../Redux/Services/Referral/Referral.api';

jest.mock('../../../Components/Pages/GiftYnvisto/GiftYnvisto.utils', () => ({
  getInfoReferral: jest.fn(() => new Promise(((resolve) => resolve({
    redeemedInvites: 12,
    remainingInvites: 1,
  } as IReferralResponse)))),

}));

test('Renders GiftYnvisto component and match to previous snapshot', async () => {
  const component = mount(<GiftYnvisto />);

  await act(
    () => new Promise((resolve) => {
      setImmediate(async () => {
        resolve();
      });
    }),
  );
  component.update();
  expect(component).toMatchSnapshot();
});
