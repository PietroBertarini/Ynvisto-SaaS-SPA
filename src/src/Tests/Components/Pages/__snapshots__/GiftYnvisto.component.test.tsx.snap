// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders GiftYnvisto component and match to previous snapshot 1`] = `
<GiftYnvisto>
  <styled.div>
    <div
      className="sc-hKFyIo cnVgug"
    >
      <Referral
        buttonTitle="Clique para copiar seu convite"
        onClick={[Function]}
        svg={
          <ForwardRef(SvgGift)
            style={
              Object {
                "height": "50vh",
              }
            }
          />
        }
        texts={
          Array [
            "Você conhece algum investidor que está precisando de uma ajuda com a tributação dos seus investimentos?",
            "Se sim, dê de presente 1 mês grátis de Ynvisto! Caso ele não tenha experimentado a plataforma ainda, o seu convite vale uma oferta especial de 1 mês grátis.",
            "Você sabia que já salvou 12 pessoas do inferno tributário brasileiro?",
          ]
        }
        title="Ajude outros investidores dando a Ynvisto de presente!"
        underButtonText="Você tem direito à mais 1 convites com mês grátis"
      >
        <styled.div>
          <div
            className="sc-dlnjPT dNkSvg"
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-header-1"
            >
              <ForwardRef(Typography)
                className="makeStyles-header-1"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
              >
                <p
                  className="MuiTypography-root makeStyles-header-1 MuiTypography-body1"
                >
                  Ajude outros investidores dando a Ynvisto de presente!
                </p>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <div
              className="svgArea"
            >
              <ForwardRef(SvgGift)
                style={
                  Object {
                    "height": "50vh",
                  }
                }
              >
                <svg
                  style={
                    Object {
                      "height": "50vh",
                    }
                  }
                >
                  gift.svg
                </svg>
              </ForwardRef(SvgGift)>
            </div>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text-2"
              key="Você conhece algum investidor que está precisando de uma ajuda com a tributação dos seus investimentos?"
            >
              <ForwardRef(Typography)
                className="makeStyles-text-2"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
              >
                <p
                  className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
                >
                  Você conhece algum investidor que está precisando de uma ajuda com a tributação dos seus investimentos?
                </p>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text-2"
              key="Se sim, dê de presente 1 mês grátis de Ynvisto! Caso ele não tenha experimentado a plataforma ainda, o seu convite vale uma oferta especial de 1 mês grátis."
            >
              <ForwardRef(Typography)
                className="makeStyles-text-2"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
              >
                <p
                  className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
                >
                  Se sim, dê de presente 1 mês grátis de Ynvisto! Caso ele não tenha experimentado a plataforma ainda, o seu convite vale uma oferta especial de 1 mês grátis.
                </p>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text-2"
              key="Você sabia que já salvou 12 pessoas do inferno tributário brasileiro?"
            >
              <ForwardRef(Typography)
                className="makeStyles-text-2"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
              >
                <p
                  className="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
                >
                  Você sabia que já salvou 12 pessoas do inferno tributário brasileiro?
                </p>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <div
              className="buttonArea"
            >
              <YnvistoButton
                onClick={[Function]}
                title="Clique para copiar seu convite"
                underButtonText="Você tem direito à mais 1 convites com mês grátis"
              >
                <div
                  className="makeStyles-main-5"
                >
                  <WithStyles(ForwardRef(Button))
                    className="makeStyles-button-3 makeStyles-button-6"
                    onClick={[Function]}
                    variant="outlined"
                  >
                    <ForwardRef(Button)
                      className="makeStyles-button-3 makeStyles-button-6"
                      classes={
                        Object {
                          "colorInherit": "MuiButton-colorInherit",
                          "contained": "MuiButton-contained",
                          "containedPrimary": "MuiButton-containedPrimary",
                          "containedSecondary": "MuiButton-containedSecondary",
                          "containedSizeLarge": "MuiButton-containedSizeLarge",
                          "containedSizeSmall": "MuiButton-containedSizeSmall",
                          "disableElevation": "MuiButton-disableElevation",
                          "disabled": "Mui-disabled",
                          "endIcon": "MuiButton-endIcon",
                          "focusVisible": "Mui-focusVisible",
                          "fullWidth": "MuiButton-fullWidth",
                          "iconSizeLarge": "MuiButton-iconSizeLarge",
                          "iconSizeMedium": "MuiButton-iconSizeMedium",
                          "iconSizeSmall": "MuiButton-iconSizeSmall",
                          "label": "MuiButton-label",
                          "outlined": "MuiButton-outlined",
                          "outlinedPrimary": "MuiButton-outlinedPrimary",
                          "outlinedSecondary": "MuiButton-outlinedSecondary",
                          "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                          "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                          "root": "MuiButton-root",
                          "sizeLarge": "MuiButton-sizeLarge",
                          "sizeSmall": "MuiButton-sizeSmall",
                          "startIcon": "MuiButton-startIcon",
                          "text": "MuiButton-text",
                          "textPrimary": "MuiButton-textPrimary",
                          "textSecondary": "MuiButton-textSecondary",
                          "textSizeLarge": "MuiButton-textSizeLarge",
                          "textSizeSmall": "MuiButton-textSizeSmall",
                        }
                      }
                      onClick={[Function]}
                      variant="outlined"
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        className="MuiButton-root MuiButton-outlined makeStyles-button-3 makeStyles-button-6"
                        component="button"
                        disabled={false}
                        focusRipple={true}
                        focusVisibleClassName="Mui-focusVisible"
                        onClick={[Function]}
                        type="button"
                      >
                        <ForwardRef(ButtonBase)
                          className="MuiButton-root MuiButton-outlined makeStyles-button-3 makeStyles-button-6"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component="button"
                          disabled={false}
                          focusRipple={true}
                          focusVisibleClassName="Mui-focusVisible"
                          onClick={[Function]}
                          type="button"
                        >
                          <button
                            className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-3 makeStyles-button-6"
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            type="button"
                          >
                            <span
                              className="MuiButton-label"
                            >
                              Clique para copiar seu convite
                            </span>
                            <WithStyles(memo)
                              center={false}
                            >
                              <ForwardRef(TouchRipple)
                                center={false}
                                classes={
                                  Object {
                                    "child": "MuiTouchRipple-child",
                                    "childLeaving": "MuiTouchRipple-childLeaving",
                                    "childPulsate": "MuiTouchRipple-childPulsate",
                                    "ripple": "MuiTouchRipple-ripple",
                                    "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                    "rippleVisible": "MuiTouchRipple-rippleVisible",
                                    "root": "MuiTouchRipple-root",
                                  }
                                }
                              >
                                <span
                                  className="MuiTouchRipple-root"
                                >
                                  <TransitionGroup
                                    childFactory={[Function]}
                                    component={null}
                                    exit={true}
                                  />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Button)>
                  </WithStyles(ForwardRef(Button))>
                  <h5
                    className="makeStyles-underText-4"
                  >
                    Você tem direito à mais 1 convites com mês grátis
                  </h5>
                </div>
              </YnvistoButton>
            </div>
          </div>
        </styled.div>
      </Referral>
    </div>
  </styled.div>
</GiftYnvisto>
`;
